### PR TITLE
Add dark mode support

### PR DESCRIPTION
**Pull request by lalitpatel**
### Summary
Respect the browsers / base OS's dark mode preference.

### Implementation
* Use `@media (prefers-color-scheme: dark)` to set the theme for the web app automatically (auto everything!!)
* Use [bulma-prefers-dark](https://github.com/jloh/bulma-prefers-dark) as the base theme for dark mode.
* Use [Google Material Icons](https://material.io/resources/icons/?style=baseline) font instead of images to render Replay Controls and Back button. 
* Use Sass and convert styles to `.scss` format.

### Not doing
Manual toggle for turning dark mode on / off. Will do it later with navbar UI refactoring.

### Screenshots
dark

light

### CODE REVIEW

**Comment by HSZemi** [comment]
I must confess, I am a bit lost… How can I check out the dark mode locally? So far, I have not been able to get it activated.

**Comment by lalitpatel** [comment]
> I must confess, I am a bit lost… How can I check out the dark mode locally? So far, I have not been able to get it activated.

This preference is OS dependent. The browser propagate the preference of user set at the OS level.

It is also possible to simulate this at the browser level:

- For [Firefox](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme), you can set a config value by going to `about:config` and creating a new numeric preference `ui.systemUsesDarkTheme` set to `1`.
- For Chrome, you can go to the Dev Tools > Command Center > Emulate CSS  prefers-color-scheme: dark ([instructions](https://dev.to/codepo8/quick-developer-tools-tip-simulating-dark-light-colour-mode-1cpg))

File: src/components/PresetEditor/PresetEditorTurn.tsx [hunk 1]
```diff
@@ -10,8 +10,7 @@ interface IProps {
 }
 
 export const PresetEditorTurn = ({index, turn, onValueChange}: IProps) =>
-    <div className="columns is-mobile has-background-light" key={'turn-' + index}
-         style={{borderBottom: '1px solid #ddd'}}>
+    <div className="columns is-mobile preset-editor-row" key={'turn-' + index}>
```
Comment (HSZemi) [inline src/components/PresetEditor/PresetEditorTurn.tsx:13]: That seems to be a bit darker than the previous background colour, which makes the checkboxes stand out. Is that on purpose?
image

Comment (HSZemi) [reply src/components/PresetEditor/PresetEditorTurn.tsx:13]: Similar issue with the dark theme
image

Comment (lalitpatel) [reply src/components/PresetEditor/PresetEditorTurn.tsx:13]: This is not intentional. It is a side effect of some other style I changed. I will correct it.

Comment (lalitpatel) [reply src/components/PresetEditor/PresetEditorTurn.tsx:13]: Fixed.

**Comment by HSZemi** [comment]
Thank you :yellow_heart:

**Pull request merged**